crypto/fipsmodule/rand/rand.c

```diff
@@ -539,6 +554,12 @@ void RAND_bytes_with_additional_data(uint8_t *out, size_t out_len,
 
   OPENSSL_cleanse(additional_data, 32);
 
+  CRYPTO_get_snapsafe_generation(&snapsafe_generation_id);
```

Comment:
Should this check the return value of CRYPTO_get_snapsafe_generation?

Reply:
Good point.  We only really care about this when snapsafe detection is active.  (Although when it's inactive, the generation value wouldn't change anyways.) I'll make this logic also conditioned on `CRYPTO_get_snapsafe_generation` returning 1.